app/org/sagebionetworks/bridge/services/HealthDataService.java

```diff
@@ -2,24 +2,74 @@
 
 import java.util.List;
 
-import org.sagebionetworks.bridge.models.GuidVersionHolder;
-import org.sagebionetworks.bridge.models.healthdata.HealthDataKey;
+import com.google.common.base.Strings;
+import org.joda.time.LocalDate;
+import org.joda.time.format.ISODateTimeFormat;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import org.sagebionetworks.bridge.dao.HealthDataDao;
+import org.sagebionetworks.bridge.exceptions.BadRequestException;
+import org.sagebionetworks.bridge.exceptions.InvalidEntityException;
 import org.sagebionetworks.bridge.models.healthdata.HealthDataRecord;
+import org.sagebionetworks.bridge.validators.HealthDataRecordValidator;
+import org.sagebionetworks.bridge.validators.Validate;
+
+/** Service handler for health data APIs. */
+@Component
+public class HealthDataService {
+    private HealthDataDao healthDataDao;
+
+    /** Health data DAO. This is configured by Spring. */
+    @Autowired
+    public void setHealthDataDao(HealthDataDao healthDataDao) {
+        this.healthDataDao = healthDataDao;
+    }
+
+    /**
+     * Creates and persists a health data record, using the passed in prototype object as the base. This method is
+     * generally used by worker apps as part of upload unpacking.
+     *
+     * @param record
+     *         the prototype record to create from and persist, must be non-null and have valid fields
+     * @return the record ID of the persisted record
+     */
+    public String createRecord(HealthDataRecord record) {
+        // validate record
+        if (record == null) {
+            throw new InvalidEntityException(String.format(Validate.CANNOT_BE_NULL, "HealthDataRecord"));
+        }
+        Validate.entityThrowingException(HealthDataRecordValidator.INSTANCE, record);
+
+        // TODO: validate health code, schema ID
 
-public interface HealthDataService {
+        // call through to DAO
+        return healthDataDao.createRecord(record);
+    }
 
-    public List<HealthDataRecord> appendHealthData(HealthDataKey key, List<HealthDataRecord> records);
-    
-    public List<HealthDataRecord> getAllHealthData(HealthDataKey key);
-    
-    public List<HealthDataRecord> getHealthDataByDateRange(HealthDataKey key, long startDate, long endDate);
-    
-    public HealthDataRecord getHealthDataRecord(HealthDataKey key, String guid);
+    /**
+     * Returns a list of all health data records with the given upload date. THis method is generally called by
+     * worker apps as part of data export.
+     *
+     * @param uploadDate
+     *         an upload date in YYYY-MM-DD format, must be non-null, non-empty, and represent a valid date
+     * @return list of health data records
+     */
+    public List<HealthDataRecord> getRecordsForUploadDate(String uploadDate) {
+        // validate upload date
+        if (Strings.isNullOrEmpty(uploadDate)) {
```

Comment:
FWIW, this method allows empty strings like "   ", whereas the Apache version isNotBlank() does not.